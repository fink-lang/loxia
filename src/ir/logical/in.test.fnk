{fink2lir} = import '../../testing/generate.fnk'
{skip, describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'


describe 'in', fn:
  it 'compiles with default runtime', fn:
    expect
      fink2lir "
        foo = 1 in [1, 2, 3]
        bar = '2' in '1234'
        spam = key in {foo: bar}
        shrub = item in ni
      "
      to_match_snapshot


  skip.it 'compiles as partial', fn:
    expect
      fink2lir "
        ? in [1, 2, 3]
        foo in ?
      "
      to_match_snapshot


