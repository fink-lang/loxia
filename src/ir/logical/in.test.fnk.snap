// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in compiles as partial 1`] = `
"import { _in_ } from \\"@fink/js-interop/runtime.js\\";

ˆpartial => _in_(ˆpartial, [1, 2, 3]);

ˆpartial => _in_(foo, ˆpartial);"
`;

exports[`in compiles with default runtime 1`] = `
"
rec >> exports_0:
  int 1 >> left_0:
    lst >> lst_2:
      int 1 >> item_2:
        lst_a lst_2, item_2 >> lst_1:
          int 2 >> item_1:
            lst_a lst_1, item_1 >> lst_0:
              int 3 >> item_0:
                lst_a lst_0, item_0 >> right_0:
                  in left_0, right_0 >> foo_0:
                    str 'foo' >> key_0
                      rec_s exports_0, key_0, foo_0 >> exports_1:
                        str '2' >> str_0
                          strt str_0 >> left_1:
                            str '1234' >> str_1
                              strt str_1 >> right_1:
                                in left_1, right_1 >> bar_0:
                                  str 'bar' >> key_1
                                    rec_s exports_1, key_1, bar_0 >> exports_2:
                                      = key >> left_2:
                                        rec >> rec_0:
                                          str 'foo' >> key_2
                                            = bar_0 >> val_0:
                                              rec_s rec_0, key_2, val_0 >> right_2:
                                                in left_2, right_2 >> spam_0:
                                                  str 'spam' >> key_3
                                                    rec_s exports_2, key_3, spam_0 >> exports_3:
                                                      = item >> left_3:
                                                        = ni >> right_3:
                                                          in left_3, right_3 >> shrub_0:
                                                            str 'shrub' >> key_4
                                                              rec_s exports_3, key_4, shrub_0 >> exports_4:
                                                                lst >> drctvs_0:
                                                                  mod exports_4, drctvs_0 >> mod_0:"
`;
